src/contracts/core/StakeRootCompendium.sol

```diff
@@ -17,52 +18,81 @@ contract StakeRootCompendium is StakeRootCompendiumStorage {
         uint256 _minBalanceThreshold,
         uint256 _minPrepaidProofs,
         address _verifier,
-        bytes32 _imageId    
-    ) StakeRootCompendiumStorage(_delegationManager, _avsDirectory, _allocationManager, _minBalanceThreshold, _minPrepaidProofs, _verifier, _imageId) {}
-
-    function initialize(address _owner, address _rootConfirmer, uint32 _proofIntervalSeconds, uint96 _maxTotalCharge, uint96 _chargePerStrategy, uint96 _chargePerOperatorSet) public initializer {
+        bytes32 _imageId
+    )
+        StakeRootCompendiumStorage(
+            _delegationManager,
+            _avsDirectory,
+            _allocationManager,
+            _minBalanceThreshold,
+            _minPrepaidProofs,
+            _verifier,
+            _imageId
+        )
+    {}
+
+    function initialize(
+        address _owner,
+        address _rootConfirmer,
+        uint32 _proofIntervalSeconds,
+        uint96 _maxChargePerProof,
+        uint96 _chargePerStrategy,
+        uint96 _chargePerOperatorSet
+    ) public initializer {
         __Ownable_init();
         _transferOwnership(_owner);
 
         rootConfirmer = _rootConfirmer;
 
-        maxTotalCharge = _maxTotalCharge;
-        proofIntervalSeconds = _proofIntervalSeconds;
-        chargePerOperatorSet = _chargePerOperatorSet;
-        chargePerStrategy = _chargePerStrategy;
+        chargeParams = ChargeParams({
+            chargePerOperatorSet: _chargePerOperatorSet,
+            chargePerStrategy: _chargePerStrategy,
+            maxChargePerProof: _maxChargePerProof
+        });
+
+        cumulativeChargeParams.proofIntervalSeconds = _proofIntervalSeconds;
```

Comment:
move `proofIntervalSeconds` to `ChargeParams` imo

Reply:
I see the abstraction point, but not for data packing. I think this balances abstraction with gas optimization. 

The `cumulativeChargeParams` are used for balance updates while the `chargeParams` are only relevant when updating the `chargePerProof` (or "`totalCharge`") does not use the `proofIntervalSeconds` which is only relevant for rate limiting proof verification as well as updating the `cumulativeCharge` which is why it was included in `cumulativeChargeParams`

Reply:
i see. im fine with this